Foundation/FileHandle.swift

```diff
@@ -425,6 +433,15 @@ open class FileHandle : NSObject, NSSecureCoding {
         }
     }
 #endif
+    
+    internal init?(fileSystemRepresentation: UnsafePointer<Int8>, flags: Int32, createMode: Int) {
```

Comment:
I don't think that we should do this.  The problem is that on Windows, you need to use UTF16 (or Unicode) for the filenames for the FSR otherwise you will not be able to open all files.

Reply:
The methods `_compareFiles()` and `contentEquals()` does not support Windows platform, yet. We will cross the bridge when we get to it. As for as this PR concern it seems to me, it is safe to wrap this `init` method under `#if !os(Windows)`. What do you think @compnerd?

Reply:
The SR was filed specifically to enable Windows support, so I think that it should be part of this change.  Using the `FileHandle` was the way around the FSR conversions.

Reply:
Sounds good. I will update the code to support Windows as well.

Reply:
@compnerd  Thanks for the suggestions. I have updated the init method. Now it supports windows as well. Please do a round of review when you get a chance.

Reply:
Maybe a better solution would be to add an internal typealias for FSR
```
#if os(windows)
internal typealias _FileSystemRepresentation = UnsafePointer<UInt16>
#else
internal typealias _FileSystemRepresentation = UnsafePointer<Int8>
#endif
```

And use that in the signatures for these internal functions. I guess `FileManager.fileSystemRepresentation(withPath path: String) -> UnsafePointer<Int8>` is technically broken on Windows anyway but can probably be ignored for now.

Reply:
Hmm, we don't want to break the public interfaces, I think that we should convert across the encodings.